agent/exec/containerd/controller.go

```diff
@@ -0,0 +1,364 @@
+package containerd
+
+import (
+	"fmt"
+
+	"github.com/containerd/containerd/api/types/container"
+	"github.com/docker/swarmkit/agent/exec"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/log"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type controller struct {
+	task    *api.Task
+	adapter *containerAdapter
+	closed  chan struct{}
+	err     error
+
+	pulled     chan struct{} // closed after pull
+	cancelPull func()        // cancels pull context if not nil
+	pullErr    error         // pull error, protected by close of pulled
+}
+
+var _ exec.Controller = &controller{}
+
+func newController(conn *grpc.ClientConn, containerDir string, task *api.Task, secrets exec.SecretGetter) (exec.Controller, error) {
+	adapter, err := newContainerAdapter(conn, containerDir, task, secrets)
+	if err != nil {
+		return nil, err
+	}
+
+	return &controller{
+		task:    task,
+		adapter: adapter,
+		closed:  make(chan struct{}),
+	}, nil
+}
+
+// Update tasks a recent task update and applies it to the container.
```

Comment:
Seems like this comment came from `agent/exec/dockerapi/controller.go`. I'll change that too.